DSA/2.2 Searching & Sorting Algorithms/Infinite_Array_Binary_Search.md

```diff
@@ -0,0 +1,103 @@
+# Search an element in a sorted and infinite array
+
+### **Description**
+You are given a sorted and infinite array A[] and an element K. You need to search for the element K in the array. If found, return the index of the element, else return -1.
+
+
+
+#### EXAMPLE 1:
+Input: A[] = {1,2,4,8,11,13,15,17,19,28,39,103,123,140,2040,…}, K = 17
+
+Output: 6
+
+#### EXAMPLE 2:
+Input: A[] = {1,3,5,8,10,20,25,30,67,93,159,192,350,1230,1341,4533,…}, K = 23
+
+Output: -1
+
+***
+### Approches
+
+* BRUTE FORCE 
+* EFFICENT SOLUTION
+
+### Brute Force approach : Using linear Search
+
+In the infinite array, we don’t know the upper bound to apply the binary search. So simple solution we could think to start searching for K linearly from index 0  until you you find an element equal to K, then return the index. If you find an element greater than K, then return -1.
+
+### Efficent Solution : Binary Search
+
+In this approach, we increase the interval size by an exponential order of 2. First set the index at position 2, then 4, then 8, then 16 and so on. Again do the binary search from position LEFT to RIGHT where item is in between. We call this approach exponential search which helps us to track the upper bound quickly in comparison to the previous approach.
```

Comment:
You can use points to describe your algorithm

Reply:
> You can use points to describe your algorithm

THanks for reviewing , I will do the changes and will update it